pkg/glusterutils/exporterd.go

```diff
@@ -13,25 +14,14 @@ var (
 	peerIDPattern = regexp.MustCompile("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}")
 )
 
-// Peers return a list of peers
-func Peers(config *Config) ([]Peer, error) {
-	setDefaultConfig(config)
-
-	if config.GlusterMgmt == MgmtGlusterd2 {
-		return peersGD2(config)
-	}
-
-	return peersGD1(config)
-}
-
 // IsLeader returns true or false based on whether the node is the leader of the cluster or not
-func IsLeader(config *Config) (bool, error) {
-	setDefaultConfig(config)
-	peerList, err := Peers(config)
+func (g GD1) IsLeader(config *Config) (bool, error) {
```

Comment:
- config parameter can be removed since already available as `g.config`
- change `(g GD1)` to `(g *GD1)`

Reply:
We also need `func (g *GD2) IsLeader() (bool, error)`

```
func (g *GD1) IsLeader() (bool, error) {
	peerList, err := Peers(config)
	if err != nil {
		return false, err
	}
	peerID, err := LocalPeerID(config)
	if err != nil {
		return false, err
	}

	var maxPeerID string
	//This for loop iterates among all the peers and finds the peer with the maximum UUID (lexicographically)
	for i, pr := range peerList {
		if pr.Online {
			if peerList[i].ID > maxPeerID {
				maxPeerID = peerList[i].ID
			}
		}
	}
	//Checks and returns true if maximum peerID is equal to the local peerID
	if maxPeerID == peerID {
		return true, nil
	}
	return false, nil
}

func (g *GD2) IsLeader() (bool, error) {
	peerList, err := Peers(config)
	if err != nil {
		return false, err
	}
	peerID, err := LocalPeerID(config)
	if err != nil {
		return false, err
	}
	//The following lines checks and returns true if the local PeerID is equal to that of the first PeerID in the list
	for _, pr := range peerList {
		if pr.Online {
			if peerID == peerList[0].ID {
				return true, nil
			}
			return false, nil
		}
	}
	// This would imply none of the peers are online and return false
	return false, nil
}
```